addons/mail/models/mail_message.py

```diff
@@ -842,8 +900,15 @@ def _notify(self, layout=False, force_send=False, send_after_commit=True, values
         if not self._context.get('mail_notify_author', False) and self_sudo.author_id:
             partners_sudo = partners_sudo - self_sudo.author_id
 
-        # update message, with maybe custom values
+        # list channels and partner by notification type
+        email_channels = channels_sudo.filtered(lambda channel: channel.email_send)
+        notif_partners = partners_sudo.filtered(lambda partner: 'inbox' in partner.mapped('user_ids.notification_type'))
+        email_partner = partners_sudo - notif_partners
+
+        #update message, with maybe custom values
         message_values = {}
+        if email_partner:
+            message_values = {'layout': layout, 'layout_values': repr(values)}
```

Comment:
use `json.dumps()` unless there is a good reason to use repr/safe_eval, and in that case document it